public/app/features/explore/Logs/Logs.tsx

```diff
@@ -248,319 +242,334 @@ class UnthemedLogs extends PureComponent<Props, State> {
-    });
-  };
-
-  onClickScan = (event: React.SyntheticEvent) => {
-    event.preventDefault();
-    if (this.props.onStartScanning) {
-      this.props.onStartScanning();
-      reportInteraction('grafana_explore_logs_scanning_button_clicked', {
-        type: 'start',
-        datasourceType: this.props.datasourceType,
+    setHiddenLogLevels(hiddenLogLevels);
+  }, []);
+
+  const onToggleLogsVolumeCollapse = useCallback(
+    (collapsed: boolean) => {
+      props.onSetLogsVolumeEnabled(!collapsed);
+      reportInteraction('grafana_explore_logs_histogram_toggle_clicked', {
+        datasourceType: props.datasourceType,
+        type: !collapsed ? 'open' : 'close',
       });
-    }
-  };
-
-  onClickStopScan = (event: React.SyntheticEvent) => {
-    event.preventDefault();
-    if (this.props.onStopScanning) {
-      this.props.onStopScanning();
-    }
-  };
-
-  showField = (key: string) => {
-    const index = this.state.displayedFields.indexOf(key);
-
-    if (index === -1) {
-      this.setState((state) => {
-        return {
-          displayedFields: state.displayedFields.concat(key),
-        };
-      });
-    }
-  };
-
-  hideField = (key: string) => {
-    const index = this.state.displayedFields.indexOf(key);
-    if (index > -1) {
-      this.setState((state) => {
-        return {
-          displayedFields: state.displayedFields.filter((k) => key !== k),
-        };
-      });
-    }
-  };
+    },
+    [props]
+  );
+
+  const onClickScan = useCallback(
+    (event: React.SyntheticEvent) => {
+      event.preventDefault();
+      if (props.onStartScanning) {
+        props.onStartScanning();
+        reportInteraction('grafana_explore_logs_scanning_button_clicked', {
+          type: 'start',
+          datasourceType: props.datasourceType,
+        });
+      }
+    },
+    [props]
+  );
+
+  const onClickStopScan = useCallback(
+    (event: React.SyntheticEvent) => {
+      event.preventDefault();
+      if (props.onStopScanning) {
+        props.onStopScanning();
+      }
+    },
+    [props]
+  );
 
-  clearDetectedFields = () => {
-    this.setState((state) => {
-      return {
-        displayedFields: [],
-      };
-    });
-  };
+  const showField = useCallback(
+    (key: string) => {
+      const index = displayedFields.indexOf(key);
 
-  onCloseContext = () => {
-    this.setState({
-      contextOpen: false,
-      contextRow: undefined,
+      if (index === -1) {
+        setDisplayedFields(displayedFields.concat(key));
+      }
+    },
+    [displayedFields]
+  );
+
+  const hideField = useCallback(
+    (key: string) => {
+      const index = displayedFields.indexOf(key);
+      if (index > -1) {
+        setDisplayedFields(displayedFields.filter((k) => key !== k));
+      }
+    },
+    [displayedFields]
+  );
+
+  const clearDetectedFields = useCallback(() => {
+    setDisplayedFields([]);
+  }, []);
+
+  const [onCloseCallback, setOnCloseCallback] = useState<() => void>(() => {});
```

Comment:
This can also just be a ref, as changing the value has no rendering consequence.

Reply:
Addressed in https://github.com/grafana/grafana/pull/87808/commits/55b82654ab0c6dbffef5a3164c486829a288a301